lib/services/consumed/status-bar/status-bar.js

```diff
@@ -45,19 +42,13 @@ export class StatusBarConsumer {
     return disposable;
   }
 
-  showKernelCommands({
-    kernel,
-    markers
-  }: {
-    kernel: Kernel,
-    markers: MarkerStore
-  }) {
+  showKernelCommands(store: Store, handleKernelCommand: Function) {
     let signalListView = this.signalListView;
     if (!signalListView) {
-      signalListView = new SignalListView(kernel);
-      signalListView.onConfirmed = (kernelCommand: { command: string }) =>
-        this.handleKernelCommand(kernelCommand, { kernel, markers });
+      signalListView = new SignalListView(store, handleKernelCommand);
       this.signalListView = signalListView;
+    } else {
+      signalListView.store = store;
```

Comment:
curious as to whether this line is truly necessary as store never changes because it is global among hydrogen. If it is or isn't necessary why not update the handle kernel command too?

Reply:
uh-huh, I know you will question this line. Actually I didn't expect it when I was changing kernel to store. I though this line should be useless as store.kernel should always return the active kernel. However, this line has to be here in reality, otherwise the store.kernel is aways the initial one, i.e. the bug remains. My guess of the reason of this is that the onClick function actually gets a copy of Store instead of a reference. But I can't confirm it as I am not familiar with react.
As for the handleKernelCommand, it's not necessary to update it. It's stateless, not interior information carried.
I guess you might have to do a big refactor on the Store for improvement, to separate information and actions, maybe create a class call context or something similar IMHO.